core/src/main/scala/org/ensime/indexer/orientdb/SimpleOrient.scala

```diff
@@ -32,9 +34,77 @@ package api {
   case class VertexT[T](underlying: Vertex)
 
   trait EdgeT[+Out, +In]
+
+  sealed trait IndexT {
+    def orientKey: String
+  }
+
+  object IndexT {
+    case object Unique extends IndexT {
+      override def orientKey = "unique"
+    }
+
+    case object NotUnique extends IndexT {
+      override def orientKey = "notunique"
+    }
+
+    case object FullText extends IndexT {
+      override def orientKey = "fulltext"
+    }
+
+  }
+
+  trait Indexable[T <: Element] {
+    def aClass: Class[T]
+  }
+
+  object Indexable {
+    implicit object VertexIndexable extends Indexable[Vertex] {
+      override def aClass: Class[Vertex] = classOf[Vertex]
+    }
+    implicit object EdgeIndexable extends Indexable[Edge] {
+      override def aClass: Class[Edge] = classOf[Edge]
+    }
+  }
 }
 
 package object syntax {
+  import org.ensime.indexer.orientdb.api.{ IndexT, Indexable }
+
+  implicit class RichOrientGraph(graph: OrientExtendedGraph) {
+    private val schema = graph.getRawGraph.getMetadata.getSchema
+
+    def createIndexOn[E <: Element, T, F](idx: IndexT = IndexT.NotUnique)(
```

Comment:
this typesafe schema is pretty cool. I wonder if we should move it into a separate package. With the lucene helpers we could create a `database` package or something like that.